### PR TITLE
README: add EULA handling

### PR DESCRIPTION
**Pull request by chaws**
Added instructions on how to handle EULA, also updated cpackget usage and other forms of adding packs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
-  -v, --verbose     Set verbosiness: None (Errors only), -v (Info) and -vv (Debugging)
+Usage:
+  cpackget [command] [flags]
+
+Available Commands:
+  help        Help about any command
+  index       Updates public index
+  init        Initializes a pack root folder
+  pack        Adds/Removes Open-CMSIS-Pack packages
+  pdsc        Adds or removes Open-CMSIS-Pack packages in the local file system via PDSC files.
```
Comment (jkrech) [inline README.md:28]: Align wording for pack and pdsc Add/remove Open-CMSIS-Pack packages

File: README.md [hunk 2]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
-### Examples
+
+### Sepecifying the working pack root folder
+
+If cpackget is going to work on an existing pack root folder, there are two ways to specify it:
+
+1. `export CMSIS_PACK_ROOT=path/to/pack-root; cpackget pack add ARM.CMSIS`
+2. `cpackget --pack-root path/to/pack-root pack add ARM.CMSIS`
+
+The first example is more common because exporting the environment variables can happen only once.
```
Comment (jkrech) [inline README.md:50]: I don't think this is a good argument. In scripts you could redefine the environment variable. Is this statement relevant?

File: README.md [hunk 3]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
+
+But cpackget is also capable of creating a new pack root folder if needed. For example:
```
Comment (jkrech) [inline README.md:52]: For create a new pack root folder run:

File: README.md [hunk 4]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
+
+```
+$ cpackget init --pack-root path/to/new/pack-root https://keil.com/pack/index.pidx
+```
+
+The command will create a folder called `path/to/new/pack-root`, a few subfolders (`.Download`, `.Local`, `.Web`)
```
Comment (jkrech) [inline README.md:58]: a few subfolders -> the following subfolder and then list without brackets.

File: README.md [hunk 5]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
+and will place a copy of the index file (if specified) to `.Web/index.pidx`.
+
+If later it is needed to update the public index file, just run `cpackget index https://vendor.com/index.pidx` and
```
Comment (jkrech) [inline README.md:61]: An up-to-date index file of publicly available Open-CMSIS-Packs  can be retrieved by running:

File: README.md [hunk 6]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
+`.Web/index.pidx` will be updated accordingly.
+
+
+### Adding packs
 
 The commands below are examples on how to add packs:
```
Comment (jkrech) [inline README.md:67]: are examples on -> demonstrate how to add packs

File: README.md [hunk 7]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
-- `cpackget pack add path/to/Vendor.PackName.x.y.z.pack`
-- `cpackget pack add https://vendor.com/example/Vendor.PackName.x.y.z.pack`
 
-If the pack is specified by an URL, cpackget command will download `Vendor.PackName.x.y.z.pack` before installing it under the **Open-CMSIS-Pack** root folder.
-This folder should be specified via system environment variable named **CMSIS_PACK_ROOT**. If empty, cpackget will attempt creating a folder called `.cpackget` in the local directory and that will become the installation folder.
+* `cpackget pack add path/to/Vendor.PackName.x.y.z.pack`
```
Comment (jkrech) [inline README.md:69]: Install a pack version that is present in the file system already:

File: README.md [hunk 8]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
+* `cpackget pack add https://vendor.com/example/Vendor.PackName.x.y.z.pack`
```
Comment (jkrech) [inline README.md:70]: Install a pack version that can be downloaded using a web link.

File: README.md [hunk 9]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
+* `cpackget pack add Vendor.PackName.x.y.z`
```
Comment (jkrech) [inline README.md:71]: Install a pack version from the public package index. The download url will be looked up by the tool.

File: README.md [hunk 10]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
+* `cpackget pack add Vendor.PackName`
```
Comment (jkrech) [inline README.md:72]: Install the latest published version of a public package listed in the package index.

File: README.md [hunk 11]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
+* `cpackget pack add -f list-of-packs.txt`, the conten is simply a list of packs, one per line
```
Comment (jkrech) [inline README.md:73]: Type: conten -> content
Install the pack versions specified in the ascii file.  Each line specifies a single pack.

File: README.md [hunk 12]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
-- `cpackget pdsc add path/to/Vendor.PackName.pdsc`
+* `cpackget pdsc add path/to/Vendor.PackName.pdsc`
 
 Note that for adding packs via PDSC files is not possible to provide an URL as input. Only local files are allowed.
 
-Removing packs uses slightly different pack specification. Here are a few examples
-- `cpackget pack rm Vendor.PackName.x.y.z`: removes only the *x.y.z* version of the `Vendor.PackName` pack.
-- `cpackget pack rm Vendor.PackName`: removes all installed versions of the `Vendor.PackName` pack.
-- `cpackget pack rm --purge Vendor.PackName`: using `--purge` triggers removal of any downloaded files.
+### Bypassing End User License Agreement (EULA)
```
Comment (jkrech) [inline README.md:80]: Accepting the ... from the command line

File: README.md [hunk 13]
```diff
@@ -17,41 +17,91 @@ Please visit the [latest stable release](https://github.com/Open-CMSIS-Pack/cpac
...
+* `cpackget pack add --extract-embedded-license Vendor.PackName`
+
+The extracted license file will be placed next to the pack's. For example if Vendor.PackName.x.y.z had a licese file
+named `LICENSE.txt`, cpackget would extract it to `.Download/Vendor.PackName.x.y.z.LICENSE.txt`.
+
+### Removing packs
+
+Removing packs uses slightly different pack specification. Here are a few examples:
+
+* `cpackget pack rm Vendor.PackName.x.y.z`: removes only the *x.y.z* version of the `Vendor.PackName` pack.
```
Comment (jkrech) [inline README.md:98]: Probably should align the style. First a text line explaining what the command will do and then the command line example.

**Review by jkrech — CHANGES_REQUESTED** [review]
Could you please review my feedback and decide on appropriate changes.

File: README.md (hunk 2 above)
Comment (chaws) [reply README.md:50]: Agreee, I think the example with the environment variable should be enough.

File: README.md (hunk 3 above)
Comment (chaws) [reply README.md:52]: ack

File: README.md (hunk 4 above)
Comment (chaws) [reply README.md:58]: ack

File: README.md (hunk 5 above)
Comment (chaws) [reply README.md:61]: I reworded like this "To create a new pack root folder with an up-to-date index file of publicly available Open-CMSIS-Pack packs run:"

File: README.md (hunk 6 above)
Comment (chaws) [reply README.md:67]: ack

File: README.md (hunk 7 above)
Comment (chaws) [reply README.md:69]: ack

File: README.md (hunk 8 above)
Comment (chaws) [reply README.md:70]: ack

File: README.md (hunk 9 above)
Comment (chaws) [reply README.md:71]: ack

File: README.md (hunk 10 above)
Comment (chaws) [reply README.md:72]: ack

File: README.md (hunk 11 above)
Comment (chaws) [reply README.md:73]: ack

File: README.md (hunk 13 above)
Comment (chaws) [reply README.md]: ack

**Comment by chaws** [comment]
Hi @jkrech , applied your suggestions.

**Review by jkrech — APPROVED** [review]
LGTM

**Pull request merged**